src/helpers/sendVerificationCode.ts

```diff
@@ -1,23 +1,73 @@
 import { Dispatch } from "redux";
 
-import { sendVerificationCode as action } from "ducks/firebase";
+import {
+  SEND_VERIFICATION_CODE,
+  sendVerificationCode as action,
+} from "ducks/firebase";
+import { buildStatus } from "helpers/buildStatus";
+import { StatusType } from "types.d/Status";
 
 interface SendVerificationCodeParams {
   phoneNumber: string;
-  recaptchaVerifier: firebase.auth.RecaptchaVerifier;
   dispatch: Dispatch;
 }
 
 export async function sendVerificationCode({
   phoneNumber,
-  recaptchaVerifier,
   dispatch,
 }: SendVerificationCodeParams) {
-  const provider = new firebase.auth.PhoneAuthProvider(firebase.auth());
-  const verificationId = await provider.verifyPhoneNumber(
-    phoneNumber as string,
-    recaptchaVerifier,
-  );
+  const setStatus = buildStatus(SEND_VERIFICATION_CODE, dispatch);
+  setStatus(StatusType.loading);
 
-  dispatch(action({ provider, verificationId }));
+  try {
+    // Always recreate #recaptcha so we don't have to deal with re-rendering
+    // issues.
+    const recaptchaContainer = document.createElement("div");
+    recaptchaContainer.id = "recaptcha";
+    document.querySelector("#recaptcha")!.replaceWith(recaptchaContainer);
+
+    const recaptchaVerifier = new firebase.auth.RecaptchaVerifier("recaptcha", {
+      size: "invisible",
+    });
```

Comment:
Also the Firebase SDK will only let you use an element once for recaptcha, then that div can't be used a second time. Was that what was happening?

Reply:
Probably not. This was being called once per session. But if something was failing, we wouldn't know. Here, any exceptions will be properly displayed.